src/Modules/UngradedModules/FreeModElem.jl

```diff
@@ -216,14 +216,30 @@ function (==)(a::AbstractFreeModElem, b::AbstractFreeModElem)
   return a.coords == b.coords
 end
 
-function hash(a::AbstractFreeModElem, h::UInt)
+# A special method for the case where we can safely assume 
+# that the coordinates of elements allow hashing.
+function hash(a::AbstractFreeModElem{<:MPolyElem{<:FieldElem}}, h::UInt)
```

Comment:
Note that a field element does not necessarily have correct hashing implemented either -- though as you correctly just pointed out to me when we talked, such fields probably won't be supported by Singular either.

But on the long run we probably want a bunch of type traits to model whether hashing, normal forms, etc. are supported:
```julia
can_hash(::Type) = false
can_compute_normal_form(::Type) = false
can_hash(::Type{<:MPolyElem{T}}) where T = can_hash(T)
can_compute_normal_form(::Type{<:MPolyElem{T}}) where T = can_compute_normal_form(T)
```
etc.

(Perhaps should turn this into an issue?)